packages/drivers/odsp-driver/src/snapshotFormatConverter.ts

```diff
@@ -0,0 +1,279 @@
+        for (const [path, value] of Object.entries(tree.trees)) {
+            treeNode.addNumber(mapping.get(path));
+            this.writeTree(treeNode.addNode(), value, mapping);
+        }
+
+        if (tree.blobs) {
+            for (const [path, id] of Object.entries(tree.blobs)) {
+                treeNode.addNumber(mapping.get(path));
+                treeNode.addNumber(mapping.get(id));
+            }
+        }
+    }
+
+    /**
+     * Represents blobs in the tree.
+     * @param node - node to serialize to.
+     * @param blobs - blobs that is being serialized
+     * @param mapping - name mapping, used to map path and IDs to integer representation
+    */
+    private writeBlobsSection(node: NodeCore, blobs: Map<string, IBlob | Uint8Array>, mapping: Map<string, number>) {
+        for (const [storageBlobId, blob] of blobs) {
+            node.addNumber(mapping.get(storageBlobId));
+            if (blob instanceof Uint8Array) {
+                node.addBlob(blob);
+            } else {
+                node.addBlob(new Uint8Array(stringToBuffer(blob.contents, blob.encoding ?? "utf-8")));
+            }
+        }
+    }
+
+    /**
+     * Represents ops in the tree.
+     * @param node - node to serialize to.
+     * @param ops - ops that is being serialized
+    */
+    private writeOpsSection(node: NodeCore, ops: ISequencedDeltaOpMessage[]) {
+        ops.forEach((op) => {
+            node.addString(JSON.stringify(op));
+        });
+    }
+
+    /**
+     * Converts snapshot from binary compact representation to ODSP snapshot.
+     * @param buffer - Compact snapshot to be parsed into odsp snapshot.
+     * @returns - tree, blobs and ops from the snapshot.
+     */
+    public convertBinaryFormatToOdspSnapshot(buffer: ReadBuffer) {
+        const builder = TreeBuilder.load(buffer);
+        let ops: ISequencedDeltaOpMessage[] | undefined;
+
+        assert(builder.length === 1, "1 root should be there");
+        const root = builder.getNode(0);
+        assert(root.length >= 4 && root.length <= 5, "4 or 5 sections should be there");
+
+        this.readAndValidateHeaderSection(root.getNode(0));
+
+        const dictionary = this.readDictionarySection(root.getNode(2));
+
+        const tree = this.readTreeSection(root.getNode(1), dictionary);
+
+        const blobs = this.readBlobSection(root.getNode(3), dictionary);
+
+        if (root.length === 5) {
+            ops = this.readOpsSection(root.getNode(4));
+        }
+
+        return { tree, blobs, ops };
+    }
+
+    /**
+     * Converts ODSP snapshot format to binary compact representation.
+     * @param snapshotTree - snapshot tree to serialize
+     * @param blobs - blobs to serialize
+     * @param snapshotSeqNumber - seq number at which snapshot is created.
+     * @param ops - ops to serialize
+     * @returns - ReadBuffer - binary representation of the data.
+     */
+    public convertOdspSnapshotToCompactSnapshot(
```

Comment:
Given this class has no state, consider removing class and exporting only public functions.
I'd probably separate functionality into multiple files - one file for one conversion direction and one - for different direction (maybe 3rd - shared code, if any). With assumption that one direction will not be used in production, and thus can be eliminated by tree shaking. As right now, tree shaking process would leave this function (as it's public method of a class that is used) even if it's not used in production code.

Reply:
Removed the class.